frontend/src/components/homepage/Filter.tsx

```diff
@@ -32,39 +34,74 @@ const Filter = ({
   setIsTranslator,
   isDisabled = false,
 }: FilterProps) => {
+  const approvedLanguages = role
+    ? approvedLanguagesTranslation
+    : approvedLanguagesReview;
+
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
     setLanguage(event.target.value);
   };
+
   const handleRoleChange = (nextRole: string) => {
```

Comment:
`setLevel(1)` should be used in this function,  on `planetread+miroslavklose@uwblueprint.org`, switching from translator to reviewer while on polish and lvl4 keeps the states on polish and lvl4.  `planetread+miroslavklose@uwblueprint.org` is only proved up to polish lvl 3 though

Reply:
I think we also talked about using `setLevel(1)` on language change too on slack - correct me if im interpreting the current ux wrong

Reply:
oh yeah mb I think I interpreted what you said in slack wrong, I'll add it in
the UI issue with the radio group showing the wrong selected level still persists though - i'll include a recording